### PR TITLE
[#8769] Extract question brief and description component

### PR DESCRIPTION
**Pull request by fans2619**
Part of #8769 
Extracted question brief and description as one component.

@wkurniawan07 
Open to any comments.

### CODE REVIEW

**Comment by fsgmhoward** [comment]
Please attach screenshots for easier review by the reviewers.

**Comment by fans2619** [comment]
> Please attach screenshots for easier review by the reviewers.

There's no change to UI. The app should look exactly the same as before.

image

**Comment by wkurniawan07** [comment]
Need to extract the part in the help page as well. If that is not done, the change done here is actually not value-adding.

**Comment by fans2619** [comment]
> Need to extract the part in the help page as well. If that is not done, the change done here is actually not value-adding.

I know, but my original intention is to let someone confirm this extraction is okay before going to change the help page. I've thought about it and with the current implementation it should be able to achieve the desired behavior for the help page.

**Comment by wkurniawan07** [comment]
I can only know whether it is ok or not when I see the component being successfully reused in the help page.

**Comment by fans2619** [comment]
> I can only know whether it is ok or not when I see the component being successfully reused in the help page.

I can update the help page later. However, this extraction itself can be totally treated as a separate issue, just whose necessity may not be justified (which depends). Also, even if I make changes to the help page using the extracted component, the focus of review should still be on whether the session edit page can work normally (but not the help page), because that's the more important component.

**Comment by fans2619** [comment]
But I also agree with you upon further thinking. Any review may be wasted if it does not work for the help page. :-)

**Comment by fans2619** [comment]
The branch name is wrong. Please ignore it.

**Comment by fans2619** [comment]
Before:

image

---

After:

image

**Comment by fans2619** [comment]
@wkurniawan07 Done √

File: src/web/app/components/question-edit-form/question-edit-form.component.html [hunk 1]
```diff
@@ -59,26 +59,11 @@
...
-                                     placeholderText="More details about the question e.g. &quot;In answering the question, do consider communications made informally within the team, and formal communications with the instructors and tutors.&quot;"
-                ></tm-rich-text-editor>
-              </div>
-            </div>
+            <tm-question-edit-brief-description-form [brief]="model.questionBrief" (briefChange)="triggerModelChange('questionBrief', $event)" [isBriefDisabled]="!model.isEditable"
+                                                     briefPlaceholder="A concise version of the question e.g. &quot;How well did the team member communicate?&quot;"
+                                                     [descriptionRichText]="model.questionDescription" (descriptionChange)="triggerModelChange('questionDescription', $event)" [isDescriptionDisabled]="!model.isEditable"
+                                                     descriptionPlaceholderText="More details about the question e.g. &quot;In answering the question, do consider communications made informally within the team, and formal communications with the instructors and tutors.&quot;"
```
Comment (wkurniawan07) [inline src/web/app/components/question-edit-form/question-edit-form.component.html:65]: I don't think we need a customizable placeholder text for either brief or description.

File: src/web/app/components/question-edit-form/question-edit-form.component.html [hunk 2]
```diff
@@ -59,26 +59,11 @@
...
-              </div>
-              <div class="col-md-10">
-                <tm-rich-text-editor [richText]="model.questionDescription" (richTextChange)="triggerModelChange('questionDescription', $event)" [isDisabled]="!model.isEditable" minHeightInPx="100"
-                                     placeholderText="More details about the question e.g. &quot;In answering the question, do consider communications made informally within the team, and formal communications with the instructors and tutors.&quot;"
-                ></tm-rich-text-editor>
-              </div>
-            </div>
+            <tm-question-edit-brief-description-form [brief]="model.questionBrief" (briefChange)="triggerModelChange('questionBrief', $event)" [isBriefDisabled]="!model.isEditable"
+                                                     briefPlaceholder="A concise version of the question e.g. &quot;How well did the team member communicate?&quot;"
+                                                     [descriptionRichText]="model.questionDescription" (descriptionChange)="triggerModelChange('questionDescription', $event)" [isDescriptionDisabled]="!model.isEditable"
```
Comment (wkurniawan07) [inline src/web/app/components/question-edit-form/question-edit-form.component.html:64]: `descriptionRichText` is slightly too wordy. Just `description` should be sufficient.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 3]
```diff
@@ -193,24 +193,9 @@ <h4 *ngIf="displaySubsection(0, 9)">Setting Up Sessions</h4>
...
-        <div class="row padding-15px text-left">
-          <div class="col-md-2">
-            <div class="question-basic-info">[Optional]<br/>Description</div>
-          </div>
-          <div class="col-md-10">
-            <tm-rich-text-editor [richText]="'<p><strong>This question is </strong><span style=\'color: rgb(255, 0, 0); background-color: rgb(255, 255, 0);\'><strong>compulsory</strong></span><strong>!</strong></p>'" minHeightInPx="100"></tm-rich-text-editor>
-          </div>
-        </div>
-      </div>
+      <tm-question-edit-brief-description-form [briefPlaceholder]="'How many hours did you spend on the project?'" [isBriefDisabled]="true"
```
Comment (wkurniawan07) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:196]: I'm aware that the previous implementation is that `How many hours did you spend on the project?` is the placeholder, but it looks like the actual value rather than placeholder.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

File: src/web/app/components/question-edit-form/question-edit-form.component.html (hunk 1 above)
Comment (fans2619) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:65]: What if you want customizable placeholders later? My questions is: Why not give different placeholders to differnt types of questions?

File: src/web/app/components/question-edit-form/question-edit-form.component.html (hunk 2 above)
Comment (fans2619) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:64]: I can make the change, but it's named `descriptionRichText` because in this way we can emphasize it's rich text instead of normal string.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html (hunk 3 above)
Comment (fans2619) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:196]: Okay I can make this change.

File: src/web/app/components/question-edit-form/question-edit-form.component.html (hunk 1 above)
Comment (wkurniawan07) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:65]: We do it when we need it. Follow the YAGNI principle.

File: src/web/app/components/question-edit-form/question-edit-form.component.html (hunk 2 above)
Comment (fans2619) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:64]: It actually took me a while to figure out how it can display text with styles before I noticed it was stored as rich text.

File: src/web/app/components/question-edit-form/question-edit-form.component.html (hunk 1 above)
Comment (fans2619) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:65]: Ok...

File: src/web/app/components/question-edit-form/question-edit-form.component.html (hunk 2 above)
Comment (wkurniawan07) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:64]: "Description being rich text" is an implementation detail within the component itself (in this case, `tm-question-edit-brief-description-form`) that the caller (in this case, `tm-question-edit-form`) does not really need to care about. We do not see `@Input`s being named `*Number`, `*String`, `*[insert custom class name here]` etc.

And within the component `tm-question-edit-brief-description-form`, it is immediately obvious by the fact that the string is rendered within a `tm-rich-text-editor` component.

Comment (fans2619) [reply src/web/app/components/question-edit-form/question-edit-form.component.html:64]: Okay I understand your rationale. I'll stick to the convention here.

**Comment by fans2619** [comment]
image

**Comment by fans2619** [comment]
image

**Review by wkurniawan07 — APPROVED** [review]
LGTM, good job
